File: 백준/2번문제/최원준/Q10799.java
// https://www.acmicpc.net/problem/
package 백준;

/*


1. 아이디어 :

2. 시간복잡도 :
O( )
3. 자료구조 :


 */
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

class Bracket {
    int idx;
    char val;

    public Bracket(int idx, char val) {
        this.idx = idx;
        this.val = val;
    }
}

class Q10799 {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static void main(String[] args) throws IOException {
        String s = br.readLine();
        Stack<Bracket> stack = new Stack<>();
        int total = 0;
        int size = 0;


        for (int i=0; i<s.length(); i++) {
            char val = s.charAt(i);
            if (stack.isEmpty() || val == '(') {
                stack.push(new Bracket(i, val));
                size ++;
            } else {
                size-=1;
                total += i - stack.peek().idx == 1? size : 1;
                stack.pop();
            }
        }
        System.out.println(total);
    }
}

/*

import sys
#sys.setrecursionlimit(1000000)
# input = sys.stdin.readline
input = lambda: sys.stdin.readline().rstrip()

def solution(s):
    stack = []
    total, size = 0, 0

    for i, val in enumerate(s):
        if not stack or val == "(":
            stack.append([i, val]) # idx, value
            size+=1
        else:
            size-=1
            total += size if i - stack[-1][0] == 1 else 1
            stack.pop()

    return total



if __name__ == '__main__':
    s = input().strip()
    print(solution(s))

 */

File: 백준/2번문제/이채현/Q10799.py
bar_razor = list(input())
answer = 0
st = []

for i in range(len(bar_razor)):
    if bar_razor[i] == '(':
        st.append('(')

    else:
        if bar_razor[i-1] == '(': 
            st.pop()
            answer += len(st)

        else:
            st.pop() 
            answer += 1 

print(answer)

# 1. 아이디어 : stack 잘 몰라서.. 인터넷 찾아가면서 했습니다.
# 아이디어는, i번째 문자가 (이면 스택에 (을 쌓습니다.
# (가 나오면 (을 쌓는 걸 반복합니다.
# )가 나왔는데 이전 것이 (이라면 그것은 레이저입니다. 따라서 바로 이전 스택 문자를 pop으로 지우고, 현재까지 스택 문자 갯수를 answer에 더해줍니다.
# )가 나왔는데 이전 것이 )이라면 바로 이전 스택 문자를 pop으로 지우고 answer에 +1을 해줍니다. 막대 갯수를 더하는 것입니다.

# 2. 시간복잡도 :
# O( )
# 3. 자료구조 :
